src/sdk/pynni/nni/curvefitting_assessor/curvefitting_assessor.py

```diff
@@ -0,0 +1,97 @@
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import logging
+from nni.assessor import Assessor, AssessResult
+from .modelfactory import CurveModel
+
+logger = logging.getLogger('curvefitting_Assessor')
+
+class CurvefittingAssessor(Assessor):
+    '''
+    CurvefittingAssessor uses learning curve fitting algorithm to predict the learning curve performance in the future.
+    It stops a pending trial X at step S if the trial's forecast result at target step is convergence and lower than the
+    best performance in the history.
+    '''
+    def __init__(self, epoch_num=20, optimize_mode='maximize', start_step=6, threshold=0.9):
+        if start_step <= 0:
+            logger.warning('It\'s recommended to set start_step to a positive number')
+        # Record the target position we predict
+        self.target_pos = epoch_num
+        # Record the optimize_mode
+        if optimize_mode == 'maximize':
+            self.higher_better = True
+        elif optimize_mode == 'minimize':
+            self.higher_better = False
+        else:
+            self.higher_better = True
+            logger.warning('unrecognized optimize_mode', optimize_mode)
+        # Start forecasting when historical data reaches start step
+        self.start_step = start_step
+        # Record the compared threshold
+        self.threshold = threshold
+        # Record the best performance
+        self.set_best_performance = False
+        self.completed_best_performance = None
+        self.trial_history = []
+        logger.info('Successfully initials the curvefitting assessor')
+
+    def trial_end(self, trial_job_id, success):
+        '''
+        trial end: update the best performance of completed trial job
+        '''
+        if success:
+            if self.set_best_performance:
+                self.completed_best_performance = max(self.completed_best_performance, self.trial_history[-1])
+            else:
+                self.set_best_performance = True
+                self.completed_best_performance = self.trial_history[-1]
+            logger.info('Updated complted best performance, trial job id:', trial_job_id)
+        else:
+            logger.info('No need to update, trial job id: ', trial_job_id)
+
+    def assess_trial(self, trial_job_id, trial_history):
+        '''
+        assess whether a trial should be early stop by curve fitting algorithm
+        return AssessResult.Good or AssessResult.Bad
+        '''
+        self.trial_history = trial_history
+        curr_step = len(trial_history)
+        if curr_step < self.start_step:
+            return AssessResult.Good
+        if self.set_best_performance is False:
```

Comment:
if not self.set_best_performance...(just suggestion)

Reply:
Modified, thanks for suggestions ：）